test/e2e/aws.go

```diff
@@ -42,3 +44,36 @@ func awsConfigWithCredentials(ctx context.Context, kubeClient client.Client, aws
 		config.WithRegion(awsRegion),
 		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(keyID, secretKey, "")))
 }
+
+// getLoadBalancerARNsByTags retrieves the ARNs of Elastic Load Balancers that match the specified tags.
+// It uses the Resource Groups Tagging API to filter load balancers based on tag criteria.
+func getLoadBalancerARNsByTags(rgtClient *resourcegroupstaggingapi.Client, tags map[string]string) ([]string, error) {
+	var tagFilters []rgtTpye.TagFilter
+	for key, value := range tags {
+		tagFilters = append(tagFilters, rgtTpye.TagFilter{
+			Key:    aws.String(key),
+			Values: []string{value},
+		})
+	}
+
+	input := &resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: []string{"elasticloadbalancing:loadbalancer"},
+		TagFilters:          tagFilters,
+	}
+
+	var arns []string
+	paginator := resourcegroupstaggingapi.NewGetResourcesPaginator(rgtClient, input)
+	for paginator.HasMorePages() {
+		page, err := paginator.NextPage(context.Background())
```

Comment:
`Context` is better to be passed as a first parameter of this function, even if it's currently only `Background()`.

Reply:
Thanks for the suggestion, updated